lib/cloudcmd/lib/server/rest.js

```diff
@@ -338,25 +339,33 @@
     function copy(from, to, names, fn) {
         var error,
             tmpl    = '\r copy {{ from }} {{ to }} {{ count }}%',
-            cp      = copymitter(from, to, names);
-         
-        cp.on('error', function(e) {
-            error = e;
-            cp.abort();
-        });
-        
-        cp.on('progress', function(count) {
```

Comment:
Ideally the progress callback will be re-implemented. The command `rsync -a --info=progress2 --progress $SRC $DST` produces output like:

```sh
neo-dashboard/node_modules/path-parse/.travis.yml
     18,197,451  80%  424.41kB/s    0:00:41 (xfr#3925, ir-chk=1005/6295)
```

Giving us a source file, and a percent complete. A caveat is that `progress2` was introduced "recently": https://serverfault.com/a/441724. I am not sure what Centos 6 has available, but MacOS for example uses `rsync` 2.6.9; it is available in Centos 7.

Reply:
If we proceed with this solution we'll want to update our list of external programs to ensure that `rsync` is both installed and has a compatible version.

Reply:
FWIW the rsync node module does appear to provide progress events though perahps it is based on later versions of rsync like you have identified above?